prow/gerrit/adapter/adapter.go

```diff
@@ -506,6 +509,27 @@ func (c *Controller) handleInRepoConfigError(err error, instance string, change
 	return nil
 }
 
+// shouldSkipProcessingChange returns true when there is no new commit or relevant commands in the comment messages
+func (c *Controller) shouldSkipProcessingChange(instance string, change client.ChangeInfo, latest client.LastSyncState) bool {
+	lastUpdate, exists := latest[instance][change.Project]
+	if !exists {
+		lastUpdate = time.Now()
+	}
+
+	revision := change.Revisions[change.CurrentRevision]
+	if revision.Created.After(lastUpdate) {
+		return false
+	}
+
+	for _, message := range currentMessages(change, lastUpdate) {
+		if pjutil.TestRe.MatchString(message.Message) || pjutil.RetestRe.MatchString(message.Message) || pjutil.TestAllRe.MatchString(message.Message) {
```

Comment:
If you move this condition out to a separate function you could add unit tests for it. Something like `messageContainsTriggeringCommand()` or similar?